lib/linter/linter.js

```diff
@@ -1199,6 +1194,9 @@ function runRules(
 
 	const lintingProblems = [];
 
+	// TODO: Necessary to initialize the .parent properties of each node. Need to find a better way...
+	sourceCode.traverse();
```

Comment:
Now we'll be calling `traverse()` twice. It doesn't have much effect on our source code implementations (js, json, css, markdown), as they're all caching the steps and will just return from the cache. But for third-party implementations, this might be a breaking change or a perf hit. For example, `html-eslint` doesn't cache steps, as it probably expects  `traverse()` to be called only once.

https://github.com/yeonjuan/html-eslint/blob/91e2ce609118eb1d0b2e2078fe465776bc60bc51/packages/eslint-plugin/lib/languages/html-source-code.js#L214-L262